pom.xml

```diff
@@ -358,6 +365,43 @@
     </dependency>
   </dependencies>
 
+  <profiles>
+    <profile>
+      <id>default</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+    </profile>
+    <profile>
+      <id>travis</id>
+      <activation>
+        <property>
+          <name>env.type</name>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <version>${version.surefire}</version>
+            <executions>
+              <execution>
+                <id>default-test</id>
+                <goals>
+                  <goal>test</goal>
+                </goals>
+                <configuration>
+                  <groups>${env.type}</groups>
```

Comment:
Not really sure if that's needed. According to [the docs](http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#groups) you should be able to use `-Dgroups` when running the build, so no need for a profile, really.

Reply:
This works fine for `mvn clean test`. But when you try to use same configuration for `mvn clean install`. It won't work. It fails saying ` Cannot find default setter in class org.apache.maven.plugins.javadoc.options.Group`

Reply:
That's a very odd type (`javadoc`?) :) Does it also show up when you use latest Surefire?

Reply:
yes, it's reproducing with latest `surefire`. ok I will try to overwrite it with `jboss-parent`

Reply:
Can you figure out if that's a known issue or limitation? Since we run `install` maybe it's the failsafe side which is causing the problem?

Reply:
no it's not the case. [maven-javadoc-plugin](https://maven.apache.org/plugins/maven-javadoc-plugin/examples/group-configuration.html) has `groups` as configuration property which is for groping packages. So it has conflict with this.

We have workaround for this as follows:
```xml
<properties>
    <surefire.groups/>
  </properties>
<plugin>
 <artifactId>maven-surefire-plugin</artifactId>
    <configuration>
      <groups>${surefire.groups}</groups>
    </configuration>
</plugin>
```

So with this we run run our tests as `mvn clean install -Dsurefire.groups=RequiresDocker`